portfolio/src/main/webapp/script.js

```diff
@@ -30,8 +30,20 @@ function change() {
   }
 }
 
-function getHello() {
-  fetch('/data').then(response => response.text()).then((quote) => {
-    document.getElementById('hello-container').innerText = quote;
+function getMessages() {
+  fetch('/data').then(response => response.json()).then((messages) => {
+    const messageListElement = document.getElementById("messages-container");
+    messageListElement.innerHTML = '';
+    for (let i = 0; i < messages.length; i++) {
+        messageListElement.appendChild(
```

Comment:
The indentation for this line and the next is a bit funky since it's part of the same line.

For this for-loop, I recommend a for-of loop to avoid a needless iteration variable e.g.

```
for (let message of messages) {
  messageListElement.appendChild(createListElement(message));
}
```